src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs

```diff
@@ -52,5 +75,77 @@ private enum PowerShellProcessArchitecture
             X86,
             X64
         }
+
+        private async Task CheckPackageManagement()
+        {
+            PSCommand getModule = new PSCommand().AddCommand("Get-Module").AddParameter("ListAvailable").AddParameter("Name", "PackageManagement");
+            foreach (PSModuleInfo module in await _powerShellContextService.ExecuteCommandAsync<PSModuleInfo>(getModule))
+            {
+                // The user has a good enough version of PackageManagement
+                if(module.Version >= s_desiredPackageManagementVersion)
+                {
+                    break;
+                }
+
+                _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                var takeActionText = "Yes";
+                MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams
+                {
+                    Message = "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                    Type = MessageType.Warning,
+                    Actions = new []
+                    {
+                        new MessageActionItem
+                        {
+                            Title = takeActionText
+                        },
+                        new MessageActionItem
+                        {
+                            Title = "Not now"
+                        }
+                    }
+                });
+
+                // If the user chose "Not now" ignore it for the rest of the session.
+                if (messageAction?.Title == takeActionText)
+                {
+                    // Update PackageManagement
+                    PSCommand psCommand = new PSCommand()
+                        .AddCommand("Install-Module")
+                            .AddParameter("Name", "PackageManagement")
+                            .AddParameter("Force")
+                            .AddParameter("MinimumVersion", s_desiredPackageManagementVersion)
+                            .AddParameter("Scope", "CurrentUser")
+                            .AddParameter("AllowClobber");
```

Comment:
So apparently because PowerShell is silly.... `psCommand.Commands[0].CommandText` just prints out `Install-Module`.. which sucks.

Three options:
* leave it like that...
* add logic to PowerShellContextService to print params as well...
* use `ExecuteScriptStringAsync` async instead and put a script in here instead of using `PSCommand`...

Thoughts @SeeminglyScience @rjmholt?

Reply:
Where does this print?

Reply:
It's here. Line 688 if I messed up the URL. It crashes my phone's browser because it's so damn big

https://www.github.com/PowerShell/PowerShellEditorServices/tree/master/src%2FPowerShellEditorServices%2FServices%2FPowerShellContext%2FPowerShellContextService.cs#L688

Reply:
I'd either use `ExecuteScriptStringAsync` or just bundle the module like we do with PSReadLine.  Or just don't show it as input

Reply:
Oh sorry, I mean where does the user see the printed output? I'm inferring from above that it looks like another command shown in the console. I think it's preferable to either not show it or fudge what's shown in some way if that's possible (like print a custom string or something)

Reply:
> Oh sorry, I mean where does the user see the printed output?

Yeah in the PSIC. It just shows `Install-Module`. Nothing else.

Reply:
I've added an `InputString` to `ExecutionOptions` so that one can overwrite what gets put into History and sent to the Host.

Reply:
so instead of that, I'm shelling out to `powershell.exe`... which would mean that as long as PackageManagement wasn't loaded in the session, the user won't have to restart their session.